src/main/java/io/cdap/e2e/pages/actions/CdfNameSpaceAdminActions.java

```diff
@@ -0,0 +1,273 @@
+
+  private static final Logger logger = LoggerFactory.getLogger(CdfNameSpaceAdminActions.class);
+  public static CdfNameSpaceAdminLocators cdfNameSpaceAdminLocators;
+
+  static {
+    cdfNameSpaceAdminLocators = SeleniumHelper.getPropertiesLocators(
+        CdfNameSpaceAdminLocators.class);
+  }
+
+  public static void clickOnNameSpaceAdminTabs(String tabName, String nameSpaceName) {
+    ElementHelper.clickOnElement(
+        CdfNameSpaceAdminLocators.nameSpaceModules(tabName, nameSpaceName));
+  }
+
+  public static void openNamespaceAdminPage() {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.namespaceAdminMenu);
+  }
+
+  public static void clickOnHamburgerMenu() {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.hamburgerMenu);
+  }
+
+  public static void clickOnEditPreference() {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.editPreferencesButton);
+  }
+
+  public static void switchToNameSpace(String nameSpaceName) {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.switchNameSpace(nameSpaceName));
+  }
+
+  public static void createProfileForNamespace(String nameSpaceName) {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.createProfile(nameSpaceName));
+  }
+
+  public static void openNamespaceDropdown() {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.namespaceDropdown);
+  }
+
+  public static void addNamespaceFromHamburgerMenu() {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.addNamespace);
+  }
+
+  /**
+   * Click on Save and Close button to save preference
+   */
+  public static void clickOnSavePreference() {
+    ElementHelper.clickOnElement(CdfNameSpaceAdminLocators.clickSaveClose);
+  }
+
+  public static void selectHamburgerMenuList(String listName) {
+    String pluginPropertyDataCyAttribute = PluginPropertyUtils.getPluginPropertyDataCyAttribute(
+        listName);
+    if (pluginPropertyDataCyAttribute == null) {
+      pluginPropertyDataCyAttribute = listName;
+    }
+    ElementHelper.clickOnElement(
+        CdfNameSpaceAdminLocators.locateMenuLink(pluginPropertyDataCyAttribute));
+  }
+
+  /**
+   * Enter KeyValue Pairs For Preference Property
+   *
+   * @param preferenceProperty @data-cy attribute value of preference Property. If
+   *                           preferenceProperty is present in
+   *                           {@link ConstantsUtil#DEFAULT_DATACY_ATTRIBUTES_FILE} then its data-cy
+   *                           is fetched from it else preferenceProperty is used as it is.
+   * @param keyValuePair       Actual json KeyValue Pairs string is fetched from
+   *                           {@link ConstantsUtil#DEFAULT_PLUGIN_PROPERTIES_FILE} with
+   *                           keyValuePair as key
+   */
+  public static void enterKeyValuePreferences(String preferenceProperty, String keyValuePair) {
+    String pluginPropertyDataCyAttribute = PluginPropertyUtils.getPluginPropertyDataCyAttribute(
```

Comment:
How is plugin being referenced on namespace admin page?

Reply:
Hi Ankit , This plugin is for setting the nameSpace admin preferences which are key-value pairs present inside datacy properties file .

Reply:
`Plugin` in CDAP is used for source/sink/transform/actions: https://cdap.atlassian.net/wiki/spaces/DOCS/pages/477790917/Introduction+to+data+pipelines#Plugins

Reply:
Changed the naming to String dataCyAttribute .